cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/ProvisionResponse.java

```diff
@@ -12,25 +15,33 @@
  * Recommendations are only relevant to {@link ProvisionStatus#UNDER_PROVISIONED} and {@link ProvisionStatus#OVER_PROVISIONED}.
  */
 public class ProvisionResponse {
+  public static final String DEFAULT_RECOMMENDATION = "N/A";
   private ProvisionStatus _status;
   private final StringBuilder _recommendation;
+  private final Map<String, ProvisionRecommendation> _recommendationByRecommender;
 
   /**
    * Constructor to be used for provision statuses, for which the recommendations are relevant.
-   * Recommendation and recommender are expected to be human-readable strings.
+   * Recommender is expected to be a human-readable string.
    *
    * @param status The current provision status.
    * @param recommendation Recommended action regarding the given provision status.
    * @param recommender The source of the recommended action to be used in aggregate recommendation.
    */
-  public ProvisionResponse(ProvisionStatus status, String recommendation, String recommender) {
+  public ProvisionResponse(ProvisionStatus status, ProvisionRecommendation recommendation, String recommender) {
```

Comment:
```suggestion
  public ProvisionResponse(ProvisionStatus status, ProvisionRecommendation provisionRecommendation, String recommender) {
```
I find it a bit hard to follow this `ProvisionRecommendation recommendation` when we have another `StringBuilder _recommendation` in the same file. Maybe we can change this name?